src/core/server/http/http_server.test.ts

```diff
@@ -577,6 +577,47 @@ test('exposes route details of incoming request to a route handler', async () =>
     });
 });
 
+describe('conditional compression', () => {
+  test('disables compression when there is a referer', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.get({ path: '/', validate: false }, (context, req, res) =>
+      res.ok({ body: 'hello', headers: { 'Content-Type': 'text/html; charset=UTF-8' } })
+    );
+    registerRouter(router);
+
+    await server.start();
+    const response = await innerServer.inject({
```

Comment:
I had to use `.inject` here,  as opposed to how the other tests are using `supertest` with the `listener` or else I wasn't able to use `response.request.info`. I also wasn't able to look at `response.headers['content-encoding']` because this isn't being set by Hapi...

Reply:
I don't feel great about testing implementation details rather than the actual HTTP response. It looks like supertest allows setting headers with the `set` method. Is it overriding it? If so, could we use a different way of sending a request in this test?

Reply:
I don't either... Even if I set the request headers, the response coming back from HapiJS doesn't have the `Content-Encoding` header set, even though it ends up having it in the functional tests...

Reply:
HapiJS seems to be skipping the actual compression in this situation, and it's not immediately obvious why: https://github.com/elastic/kibana/pull/47751/files#diff-51730cbab06a430a940f78fc9e710bd7R620-R637

Reply:
Hmm, maybe not worth figuring out why since we do at least the functional test which covers the blackbox behavior.

Reply:
I can look into it further to try to understand why. I gave up more easily than I usually do because.... hapi...